src/adapter/src/catalog/open.rs

```diff
@@ -812,6 +816,85 @@ fn add_new_remove_old_builtin_items_migration(
     }
     txn.set_system_object_mappings(new_builtin_mappings)?;
 
+    // Update comments of all builtin objects
+    let builtins_with_catalog_ids = existing_builtins
+        .iter()
+        .map(|(b, m, _)| (*b, m.unique_identifier.catalog_id))
+        .chain(
+            new_builtins
+                .into_iter()
+                .map(|((b, _), (catalog_id, _))| (b, catalog_id)),
+        );
+    for (builtin, catalog_id) in builtins_with_catalog_ids {
+        match builtin {
+            Builtin::Source(source) => {
+                match source.column_comments {
+                    Some(ref column_comments) => {
+                        assert_eq!(column_comments.len(), source.desc.len());
+                        for (col, &comment) in column_comments.iter().enumerate() {
+                            txn.update_comment(
+                                CommentObjectId::Source(catalog_id),
+                                // Column indices are 1 based
+                                Some(col + 1),
+                                Some(comment.to_owned()),
+                            )?;
+                        }
+                    }
+                    None => {
+                        txn.drop_comments(&BTreeSet::from_iter([CommentObjectId::Source(
+                            catalog_id,
+                        )]))?;
+                    }
+                }
+            }
+            Builtin::View(view) => {
+                match view.column_comments {
+                    Some(ref column_comments) => {
+                        for (col, &comment) in column_comments.iter().enumerate() {
```

Comment:
Darn, missing the num_columns == num_comments `assert!` here isn't great, but we don't have a `RelationDesc` so not much we can do :/